controllers/routemonitor/routemonitor_controller_supplement.go

```diff
@@ -0,0 +1,301 @@
+		if !k8serrors.IsNotFound(err) {
+			// return unexpectedly
+			return err
+		}
+		// populate the resource with the template
+		resource := populationFunc()
+		// and create it
+		err = r.Create(ctx, resource)
+		if err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+const (
+	blackBoxNamespace  = "openshift-monitoring"
+	blackBoxName       = "blackbox-exporter"
+	blackBoxPortName   = "blackbox"
+	blackBoxPortNumber = 9115
+)
+
+var ( // cannot be a const but doesn't ever change
+	blackBoxNamespacedName = types.NamespacedName{Name: blackBoxName, Namespace: blackBoxNamespace}
+)
+
+func (r *RouteMonitorReconciler) CreateBlackBoxExporterDeployment(ctx context.Context) error {
+	resource := &appsv1.Deployment{}
+	populationDeploymentFunc := func() *appsv1.Deployment {
+		return r.templateForBlackBoxExporterDeployment()
+	}
+	// Does the resource already exist?
+	err := r.Get(ctx, blackBoxNamespacedName, resource)
+	if err != nil {
+		// If this is an unknown error
+		if !k8serrors.IsNotFound(err) {
+			// return unexpectedly
+			return err
+		}
+		// populate the resource with the template
+		resource := populationDeploymentFunc()
+		// and create it
+		err = r.Create(ctx, resource)
+		if err != nil {
+			return err
+		}
+	}
+	return nil
+}
+func (r *RouteMonitorReconciler) CreateBlackBoxExporterService(ctx context.Context) error {
+	service := &corev1.Service{}
+	populationServiceFunc := func() *corev1.Service {
+		return r.templateForBlackBoxExporterService()
+	}
+	// Does the resource already exist?
+	if err := r.Get(ctx, blackBoxNamespacedName, service); err != nil {
+		// If this is an unknown error
+		if !k8serrors.IsNotFound(err) {
+			// return unexpectedly
+			return err
+		}
+		// populate the resource with the template
+		resource := populationServiceFunc()
+		// and create it
+		if err = r.Create(ctx, resource); err != nil {
+			return err
+		}
+	}
+	return nil
+}
+func (r *RouteMonitorReconciler) CreateBlackBoxExporterResources(ctx context.Context) error {
+	if err := r.CreateBlackBoxExporterDeployment(ctx); err != nil {
+		return err
+	}
+	if err := r.CreateBlackBoxExporterService(ctx); err != nil {
+		return err
+	}
+	return nil
+}
+
+// deploymentForBlackBoxExporter returns a blackbox deployment
+func (r *RouteMonitorReconciler) templateForBlackBoxExporterDeployment() *appsv1.Deployment {
+	labels := commonTemplateLables()
+	// hardcode the replicasize for no
+	//replicas := m.Spec.Size
+	var replicas int32 = 1
+
+	dep := &appsv1.Deployment{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      blackBoxName,
+			Namespace: blackBoxNamespace,
+			Labels:    labels,
+		},
+		Spec: appsv1.DeploymentSpec{
+			Replicas: &replicas,
+			Selector: &metav1.LabelSelector{
+				MatchLabels: labels,
+			},
+			Template: corev1.PodTemplateSpec{
+				ObjectMeta: metav1.ObjectMeta{
+					Labels: labels,
+				},
+				Spec: corev1.PodSpec{
+					Containers: []corev1.Container{{
+						Image: "prom/blackbox-exporter:master",
```

Comment:
we need to move that to quay

Reply:
like, to a local quay or a pipelined thing with quay?

Reply:
cuz I can push it to my repo for now but it will not have pipelines on it